src/QirTools/pyqir/build.ps1

```diff
@@ -0,0 +1,302 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+function Get-RepoRoot {
+    git rev-parse --show-toplevel
+}
+
+. (Join-Path (Get-RepoRoot) build "utils.ps1")
+
+function Get-CommitHash {
+    # rev-parse changes length based on your git settings, use length = 9
+    # to match azure devops
+    exec { git rev-parse --short=9 HEAD }
+}
+
+function Get-TargetTriple {
+    $triple = "unknown"
+    if ($IsWindows) {
+        $triple = "x86_64-pc-windows-msvc-static"
+    }
+    elseif ($IsLinux) {
+        $triple = "x86_64-unknown-linux-gnu"
+    }
+    else {

```

Comment:
Maybe check explicitly for Mac, and fail if an unknown platform, otherwise strange bugs will appear.